core/domain/stats_services.py

```diff
@@ -535,6 +557,25 @@ def get_exploration_stats_from_model(exploration_stats_model):
         new_state_stats_mapping)
 
 
+def get_playthrough_from_model(playthrough_model):
+    """Gets a PlaythroughModel domain object from a PlaythroughModel instance.
+
+    Args:
+        playthrough_model: PlaythroughModel. Playthrough model in datastore.
+
+    Returns:
+        Playthrough. The domain object for a playthrough.
+    """
+    actions = []
+    for action_dict in playthrough_model.actions:
+        _migrate_to_latest_action_schema(action_dict)
```

Comment:
Editing a dict in-place feels wrong since we're inside of a `get` method. `get` implies that no changes will be made to the input... I'd suggest renaming the function and adding documentation to make it more clear that changes will be made?

Or maybe this clean-up code belongs somewhere else?

Reply:
@brianrodri This system of continuous migration is in place for explorations as well, and my train of thought is this: 

We need to retrieve a domain object from the datastore. For this, we need to retrieve the model instance and convert it to a domain object (hence the get_from_model() method). The datastore model instances might have different schema versions from the current one and the domain object we use has to be consistent with the latest schema version. So, it sort of made sense to do continuous migration from the model instance's schema version to the current schema version as we convert the model to a domain object with the current schema version.

I assume there was similar reasoning behind the exploration object's continuous migration too. @seanlip What do you think?

Reply:
Ah, this one is OK. @brianrodri's point holds in general but this is a well-established pattern that's used to ensure that the domain/logic layer never operates on old data. (In fact you can't even create a domain object on a dict that's got an old schema version, it just won't work. So this step is actually mandatory.)

Reply:
Acknowledged!